### ISSUE TITLE
JSDoc autocompletion requires autoClosingBrackets & showWords setting

### ISSUE THREAD

**Issue by thany** [microsoft/vscode#170667]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.74.2
- OS Version: Windows 10 22H2

Steps to Reproduce:

1. Make sure `typescript.suggest.completeJSDocs` is set to `true`.
2. Make sure `editor.autoClosingBrackets` is set to `never`.
3. Open any typescript file
4. Above any function, type `/**`

Result:

JSDoc is not autocompleted at all. Even pressing ctrl+spacebar doesn't reveal an option to do so. The (undesired) reason is because it requires `editor.autoClosingBrackets` to be set to `languageDefined`. This makes no sense, because this has nothing to do with autoclosing brackets. JSDoc aren't brackets.

Steps to Reproduce 2:

1. Make sure `typescript.suggest.completeJSDocs` is set to `true`.
2. Make sure `editor.autoClosingBrackets` is set to `languageDefined`.
3. Make sure `editor.suggest.showWords` is set to `false`.
4. Open any typescript file
5. Above any function, type `/**`

Result:

It doesn't autocomplete JSDoc, but treats it as a regular multiline comment. In other words, it auto-inserts the closing `*/` but nothing in between. Not even a ` *` on every new line. Showing 'words' in intellisense should have nothing to do with autocompleting JSDoc. JSDoc is a totally different autocompletion thing from 'words'.

In summary:

The `typescript.suggest.completeJSDocs` setting is expected to control the ability to autocomplete JSDoc, totally indepedent of both `editor.autoClosingBrackets` and `editor.suggest.showWords` settings.

Please be noted that I have only tested this with Typescript files. The same issue might occur with Javascript files.

**Comment by mjbvz** [comment]
Thanks for end detailed write up. Both of these are not ideal but are by design currently. 

The JSDoc completion items is marked as a text suggestion: https://github.com/microsoft/vscode/blob/9ac614710c23b343c90ee32c8d6c77606eb7888f/extensions/typescript-language-features/src/languageFeatures/jsDocCompletions.ts#L22

This was previously discussed in #116941. In short: we have to chose a kind and there's no perfect fit

---

For autoclosing, without the closing `*/`, the rest of the file — including the function you are annotating — looks like a comment. This was discussed on TS here: https://github.com/microsoft/TypeScript/issues/24468#issuecomment-406060080 The complexity of fixing this meant exploring this further didn't seem worthwhile.

So it's not that we require autoclosing to be enabled, it's just that you have to be inside a valid doc comment that does not comment out the function you want to annotate

**Comment by thany** [comment]
That seems like an internals problem. At least it explains why `autoClosingBrackets` is "required". One solution is give users the ability to annotate a function without first having to type `/**`.

It doesn't quite explain why `showWords` is required. Could you elaborate on that?
